eogrow/utils/map.py

```diff
@@ -6,21 +6,35 @@
 import shutil
 import subprocess
 from tempfile import NamedTemporaryFile
-from typing import List, Union
+from typing import List, Literal, Optional, Union
 
 LOGGER = logging.getLogger(__name__)
 
+GDAL_DTYPE_SETTINGS = {
+    "uint8": "-ot Byte",
+    "uint16": "-ot UInt16",
+    "int8": "-ot Byte -co PIXELTYPE=SIGNEDBYTE",
+    "int16": "-ot Int16",
+    "float32": "-ot Float32",
+}
 
-def cogify_inplace(tiff_file: str, blocksize: int = 2048, nodata: Union[None, int, float] = None) -> None:
+
+def cogify_inplace(
+    tiff_file: str,
+    blocksize: int = 2048,
+    nodata: Union[None, int, float] = None,
```

Comment:
@zigaLuksic would `Optional[Union[int, float]]` work better here, or is it the same?

Reply:
I guess similarly as in the other comment, less brackets is better

Reply:
In fact here `Optional[float]` would be the correct. When these were typed we didnt yet know the python mypy convention that `Union[int, float]` is annotated as just `float`.

Reply:
I see, will reduce, thanks

Reply:
done